src/service/message.js

```diff
@@ -1,85 +1,133 @@
-    lineBreak = '\n';
-  }
+const addMessage = (newMessage) => new Promise((resolve, rejected) => {
+  let lineBreak = '';
 
-  fs.appendFileSync('log.txt', `${lineBreak}${message}`);
+  if (fs.existsSync('log.txt')) {
+    const rl = createInterface();
+    let newId = 0;
 
-  return message;
-};
+    rl.on('line', line => {
+      const { id, message } = JSON.parse(line);
+      if (line) {
+        fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id, message })}`);
+      }
+      if (!lineBreak) { lineBreak = '\n'; }
+
+      newId = id;
+    });
+
+    rl.on('close', () => {
+      fs.unlinkSync('log.txt');
+      fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id: ++newId, message: newMessage })}`);
+      fs.renameSync('log1.txt', 'log.txt');
+      resolve(newMessage);
+    });
+  } else {
+    rejected(new Error('file not exiest'));
+  }
+});
 
 const findAll = () => {
   if (fs.existsSync('log.txt')) {
     const fileData = fs.readFileSync('log.txt', 'utf8');
 
     if (fileData) {
-      return fileData.split('\n');
+      return fileData;
     }
   }
 
   return [];
 };
 
-const findOne = id => {
+const findOne = messageId => new Promise((resolve, rejected) => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
+    const rl = createInterface();
 
-    if (fileData) {
-      const message = fileData.split('\n')[id];
+    rl.on('line', line => {
+      const { id, message } = JSON.parse(line);
 
-      if (message) {
-        return message;
+      if (id === parseInt(messageId, 10)) {
+        resolve(message);
+        rl.close();
+        rl.removeAllListeners();
       }
-    }
-  }
+    });
 
-  return null;
-};
+    rl.on('close', () => {
+      rejected(new Error('Message has not been found'));
+    });
+  } else {
+    rejected(new Error('file not exiest'));
+  }
+});
 
+const deleteMessage = messageId => new Promise((resolve, rejected) => {
+  let lineBreak = '';
 
-const deleteMessage = id => {
   if (fs.existsSync('log.txt')) {
-    const fileData = fs.readFileSync('log.txt', 'utf8');
-
-    if (fileData) {
-      const messages = fileData.split('\n');
-
-      if (id >= messages.length) { throw new Error('Message has not been found'); }
-
-      messages.splice(id, 1);
-
-      fs.writeFileSync('log.txt', messages.join('\n'));
-
-      return;
-    }
+    const rl = createInterface();
+
+    rl.on('line', line => {
+      const { id, message } = JSON.parse(line);
+
+      if (id !== parseInt(messageId, 10)) {
+        if (line) {
+          fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id, message })}`);
+        }
+        if (!lineBreak) { lineBreak = '\n'; }
+      } else {
+        resolve();
+      }
+    });
+
+    rl.on('close', () => {
+      fs.unlinkSync('log.txt');
+      fs.renameSync('log1.txt', 'log.txt');
+      rejected(new Error('Message has not been found'));
```

Comment:
File will be closed soon or later. Why throw error?

Reply:
if there is no message at the specified id, then call reject